src/features/defi/providers/fox-farming/hooks/useFoxFarming.ts

```diff
@@ -303,16 +309,27 @@ export const useFoxFarming = (
       to: contractAddress,
       value: '0',
       wallet,
+      pubKey:
+        isTrezor(wallet) && farmingAccountId ? fromAccountId(farmingAccountId).account : undefined,
     })
 
     const txid = await buildAndBroadcast({
       adapter,
       buildCustomTxInput,
-      receiverAddress: await adapter.getAddress({ accountNumber, wallet }),
+      receiverAddress: CONTRACT_INTERACTION,
```

Comment:
This was a bit weird to read (previously we had a receive address) but works as intended